### DISCUSSION TITLE
General API documentation

### DISCUSSION THREAD

**Discussion by timmeade** [NginxProxyManager/nginx-proxy-manager#3527] [General]
Is there a definitive API documentation out there?

I am prowling around the source code trying to put together the endpoints and its difficult.

The swagger doc file is incomplete.    For example it does not contain the certificates end points...

Much thanks.  Love the project.  Just trying to document it.

**Comment by timmeade** [comment]
OK I have figured most of it out.

Using the endpoints in the code:

```
https://github.com/NginxProxyManager/nginx-proxy-manager/tree/develop/backend/routes/api
```

You can then do it in the web gui and go to the "Audit Log" tab.

It will mostly show you the JSON format.   A few tests and I had it working.

Would still be nice to have a full API doc site...

Great product by the way.   Much thanks

**Comment by Erreur32** [comment]
if you need a bash script to manage TOKEN and settings via the API for Nginx Proxy Manager.

you can test  https://github.com/Erreur32/nginx-proxy-manager-API

**Comment by YoyoSailer** [comment]
Just released https://github.com/aalasolutions/nginx-proxy-manager-sdk let me know if there is any issue or feedback
